src/main/sensors/acceleration.c

```diff
@@ -497,28 +498,25 @@ void accUpdate(timeUs_t currentTimeUs, rollAndPitchTrims_t *rollAndPitchTrims)
 
     applyAccelerationTrims(accelerationTrims);
 
-    const timeDelta_t sampleDeltaUs = currentTimeUs - lastTimeSampledUs;
-    lastTimeSampledUs = currentTimeUs;
-    accumulatedTimeUs += sampleDeltaUs;
+    ++accumulatedMeasurementCount;
     for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-        accumulator[axis] += acc.accSmooth[axis] * sampleDeltaUs;
+        accumulatedMeasurements[axis] += acc.accSmooth[axis];
     }
 }
 
-bool accGetAccumulationAverage(float *accumulation)
+bool accGetAccumulationAverage(float *accumulationAverage)
 {
-    if (accumulatedTimeUs > 0) {
-        const float accumulatedTimeS = accumulatedTimeUs * 1e-6;
+    if (accumulatedMeasurementCount > 0) {
         // If we have gyro data accumulated, calculate average rate that will yield the same rotation
         for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            accumulation[axis] = accumulator[axis] / accumulatedTimeS;
-            accumulator[axis] = 0.0f;
+            accumulationAverage[axis] = accumulatedMeasurements[axis] / accumulatedMeasurementCount;
+            accumulatedMeasurements[axis] = 0.0f;
         }
-        accumulatedTimeUs = 0;
+        accumulatedMeasurementCount = 0;
         return true;
     } else {
         for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            accumulation[axis] = 0.0f;
+            accumulationAverage[axis] = 0.0f;
```

Comment:
This will cause huge bias if `accGetAccumulationAverage` is called before sample is ready.
Mathematically, function should still return  `accumulatedMeasurements[axis] / accumulatedMeasurementCount;` - 0/0, which is not zero